app/controllers/unauthorized_controller.rb

```diff
@@ -18,14 +18,14 @@ def self.call(env)
   end
 
   def respond
-    if request.path.start_with?('/api/') || request.format == :json
-      render json: {}, status: 404
```

Comment:
the old 404 was also telling users that it was unauthorized since a genuine 404 error looks different ... also routing error would look different ... since we don't have public/private things in the api getting the 401 does not tell an attacker anything ...